djangocms_bootstrap4/contrib/bootstrap4_grid/cms_plugins.py

```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+from django.conf import settings
+from django.utils.translation import ugettext_lazy as _
+
+from cms.models import CMSPlugin
+from cms.plugin_base import CMSPluginBase
+from cms.plugin_pool import plugin_pool
+
+from ...helpers import concat_classes
+from ...constants import DEVICE_SIZES
+
+from .models import (
+    Bootstrap4GridContainer,
+    Bootstrap4GridRow,
+    Bootstrap4GridColumn,
+)
+from .forms import Bootstrap4GridRowForm, Bootstrap4GridColumnForm
+
+
+class Bootstrap4GridContainerPlugin(CMSPluginBase):
+    """
+    Layout > Grid: "Container" Plugin
+    https://getbootstrap.com/docs/4.0/layout/grid/
+    """
+    model = Bootstrap4GridContainer
+    name = _('Container')
+    module = _('Bootstrap 4')
+    render_template = 'djangocms_bootstrap4/grid_container.html'
+    allow_children = True
+
+    fieldsets = [
+        (None, {
+            'fields': (
+                'container_type',
+            )
+        }),
+        (_('Advanced settings'), {
+            'classes': ('collapse',),
+            'fields': (
+                'tag_type',
+                'attributes',
+            )
+        }),
+    ]
+
+    def render(self, context, instance, placeholder):
+        classes = concat_classes([
+            instance.container_type,
+        ], instance.attributes)
+        instance.attributes['class'] = classes
+
+        return super(Bootstrap4GridContainerPlugin, self).render(
+            context, instance, placeholder
+        )
+
+
+class Bootstrap4GridRowPlugin(CMSPluginBase):
+    """
+    Layout > Grid: "Row" Plugin
+    https://getbootstrap.com/docs/4.0/layout/grid/
+    """
+    model = Bootstrap4GridRow
+    name = _('Row')
+    module = _('Bootstrap 4')
+    form = Bootstrap4GridRowForm
+    change_form_template = 'djangocms_bootstrap4/admin/grid_row.html'
+    render_template = 'djangocms_bootstrap4/grid_row.html'
+    allow_children = True
+    child_classes = ['Bootstrap4GridColumnPlugin']
+
+    fieldsets = [
+        (None, {
+            'fields': (
+                'create',
+                ('vertical_alignment', 'horizontal_alignment'),
+            )
+        }),
+        (_('Advanced settings'), {
+            'classes': ('collapse',),
+            'fields': (
+                ('tag_type', 'gutters',),
+                'attributes',
+            )
+        }),
+    ]
+
+    def save_model(self, request, obj, form, change):
+        response = super(Bootstrap4GridRowPlugin, self).save_model(request, obj, form, change)
```

Comment:
response is misleading because `.save_model()` doesn't return a response. It doesn't return anything.